### DISCUSSION TITLE
Burning Barrel Graphic

### DISCUSSION THREAD

**Discussion by Moburma** [swfans/syndwarsfx#225] [General]
This is an oddity I only just noticed. I don't think this graphic is actually used anywhere in the game, other than on the cut old version of the Anchorage map, where there are two of them on the landing next to the boat in the river. They have an appropriate flicking light effect placed on them, but there's a problem - they don't actually animate. Here's how it's meant to look:

![](https://i.ibb.co/C2x6mpz/Syndicate-Wars-Burning-Barrel.gif)

**Comment by mefistotelis** [comment]
Yeah, the `.ani` files.. haven't decided yet what to do with them.

Bullfrog evidently had a visual editor for animations, where you could add sprites, move them, and link frames into animations.
But I did not found, as of now, any traces of parts of that being available in final code.

We will need either to make visual editor, or have the files in some text format where they could be edited manually.

**Reply by Moburma** [reply]
Yes, I realised that that was what the medit file was referring to - it must have been from a sprite/animation editor, hence why it has all the animation names in.

One thing I wonder is there are a lot of sprite viewer apps for Theme Park on Github, which has a similar file structure, which might be repurposed. I just noticed that TP comes with the MEDIT-0 file too, which again has all the animation names in, so confirms its purpose in the Bullfrog games. Syndicate of course uses the same structure too, but I can't see the MEDIT file for that in the released versions I have.

EDIT

Et voilà:

![](https://i.ibb.co/2SR5pnt/Screenshot-2024-11-26-125831.png)

EDIT2

Correct palette:

![](https://i.ibb.co/2dssZ6L/Screenshot-2024-11-26-133504.png)

**Comment by Moburma** [comment]
I realised the pre-alpha demo also has an MEDIT file. Comparing that against the Beta one is interesting, comparatively not that much had changed in terms of characters/animations. It also explained what the "OFF" character I saw listed in the frames.h code was - it's just the final game scientist. One thing I didn't notice until this comparison is that the Scientist in the Beta medit actually still has some of its frames called "OFF" whatever, making it pretty clear it's the same character. Based on that I assume it probably stood for "Official" or similar, rather than "Officer".

**Comment by mefistotelis** [comment]
I looked at the animations as well. There is one pre-alpha function which can write these, and it allows to draw some conclusions:
```
void consolidate_nsta_nfra(void)
{
    struct Frame *p_src_frame;
    struct Frame *p_frame;
    ushort frm, nxfrm;
    uint n_starts, sta;
    ushort stani;

    frm = 1;
    n_starts = (nstart_ani_end - nstart_ani);
    LOGSYNC("n_starts=%d", n_starts);
    for (sta = 1; (sta < n_starts - 1) && (frm <= 2000); sta++)
    {
        stani = nstart_ani[sta];
        if (stani == 0)
            continue;
        nstart_ani[sta] = frm;
        p_src_frame = &frame2[stani];
        p_frame = &frame[frm];
        LbMemoryCopy(p_frame, p_src_frame, sizeof(struct Frame));
        frm++;
        p_frame->Next = frm;
        for (nxfrm = p_src_frame->Next; nxfrm != stani; nxfrm = p_src_frame->Next)
        {
          p_src_frame = &frame2[nxfrm];
          p_frame = &frame[frm];
          LbMemoryCopy(p_frame, p_src_frame, sizeof(struct Frame));
          frm++;
          p_frame->Next = frm;
        }
        frame[frm - 1].Next = nstart_ani[sta];
    }
    save_range_to_file("data/nsta-0.ani", nstart_ani, nstart_ani_end);
    save_range_to_file("data/nfra-0.ani", frame, frame_end);
}
```

So, this function does cleanup of the `nfra-0.ani` - removes unused frames (that is frames which are not part of animations listed in `nsta-0.ani`). 

Anyway, if that function was used, then:
* Bullfrog seem to have had a "dev" version of frames, with a lot more of them,
* The unused frames were cut, but corresponding "elements" were left intact. So based on unused elements, we can figure out or even reconstruct the cut frames.

These cut frames are just variations on existing sprites, so they won't introduce anything new. But still, interesting.
